commit 07acfc2a9349a8ce45b236c2624dad452001966b
Merge: b5f4035adfff 322728e55aa7
Author: Linus Torvalds <torvalds@linux-foundation.org>
Date:   Thu May 24 16:17:30 2012 -0700

    Merge branch 'next' of git://git.kernel.org/pub/scm/virt/kvm/kvm
    
    Pull KVM changes from Avi Kivity:
     "Changes include additional instruction emulation, page-crossing MMIO,
      faster dirty logging, preventing the watchdog from killing a stopped
      guest, module autoload, a new MSI ABI, and some minor optimizations
      and fixes.  Outside x86 we have a small s390 and a very large ppc
      update.
    
      Regarding the new (for kvm) rebaseless workflow, some of the patches
      that were merged before we switch trees had to be rebased, while
      others are true pulls.  In either case the signoffs should be correct
      now."
    
    Fix up trivial conflicts in Documentation/feature-removal-schedule.txt
    arch/powerpc/kvm/book3s_segment.S and arch/x86/include/asm/kvm_para.h.
    
    I suspect the kvm_para.h resolution ends up doing the "do I have cpuid"
    check effectively twice (it was done differently in two different
    commits), but better safe than sorry ;)
    
    * 'next' of git://git.kernel.org/pub/scm/virt/kvm/kvm: (125 commits)
      KVM: make asm-generic/kvm_para.h have an ifdef __KERNEL__ block
      KVM: s390: onereg for timer related registers
      KVM: s390: epoch difference and TOD programmable field
      KVM: s390: KVM_GET/SET_ONEREG for s390
      KVM: s390: add capability indicating COW support
      KVM: Fix mmu_reload() clash with nested vmx event injection
      KVM: MMU: Don't use RCU for lockless shadow walking
      KVM: VMX: Optimize %ds, %es reload
      KVM: VMX: Fix %ds/%es clobber
      KVM: x86 emulator: convert bsf/bsr instructions to emulate_2op_SrcV_nobyte()
      KVM: VMX: unlike vmcs on fail path
      KVM: PPC: Emulator: clean up SPR reads and writes
      KVM: PPC: Emulator: clean up instruction parsing
      kvm/powerpc: Add new ioctl to retreive server MMU infos
      kvm/book3s: Make kernel emulated H_PUT_TCE available for "PR" KVM
      KVM: PPC: bookehv: Fix r8/r13 storing in level exception handler
      KVM: PPC: Book3S: Enable IRQs during exit handling
      KVM: PPC: Fix PR KVM on POWER7 bare metal
      KVM: PPC: Fix stbux emulation
      KVM: PPC: bookehv: Use lwz/stw instead of PPC_LL/PPC_STL for 32-bit fields
      ...

diff --cc arch/powerpc/kernel/head_fsl_booke.S
index de80e0f9a2bd,89c6d6f36785..1f4434a38608
--- a/arch/powerpc/kernel/head_fsl_booke.S
+++ b/arch/powerpc/kernel/head_fsl_booke.S
@@@ -871,9 -906,40 +906,32 @@@ _GLOBAL(__setup_e500mc_ivors
  	mtspr	SPRN_IVOR36,r3
  	li	r3,CriticalDoorbell@l
  	mtspr	SPRN_IVOR37,r3
+ 
+ 	/*
+ 	 * We only want to touch IVOR38-41 if we're running on hardware
+ 	 * that supports category E.HV.  The architectural way to determine
+ 	 * this is MMUCFG[LPIDSIZE].
+ 	 */
+ 	mfspr	r3, SPRN_MMUCFG
+ 	andis.	r3, r3, MMUCFG_LPIDSIZE@h
+ 	beq	no_hv
+ 	li	r3,GuestDoorbell@l
+ 	mtspr	SPRN_IVOR38,r3
+ 	li	r3,CriticalGuestDoorbell@l
+ 	mtspr	SPRN_IVOR39,r3
+ 	li	r3,Hypercall@l
+ 	mtspr	SPRN_IVOR40,r3
+ 	li	r3,Ehvpriv@l
+ 	mtspr	SPRN_IVOR41,r3
+ skip_hv_ivors:
  	sync
  	blr
+ no_hv:
+ 	lwz	r3, CPU_SPEC_FEATURES(r5)
+ 	rlwinm	r3, r3, 0, ~CPU_FTR_EMB_HV
+ 	stw	r3, CPU_SPEC_FEATURES(r5)
+ 	b	skip_hv_ivors
  
 -/*
 - * extern void giveup_altivec(struct task_struct *prev)
 - *
 - * The e500 core does not have an AltiVec unit.
 - */
 -_GLOBAL(giveup_altivec)
 -	blr
 -
  #ifdef CONFIG_SPE
  /*
   * extern void giveup_spe(struct task_struct *prev)